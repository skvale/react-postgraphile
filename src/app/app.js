import React from 'react'
import PropTypes from 'prop-types'
import gql from 'graphql-tag'
import { connect } from 'react-redux'
import { graphql } from 'react-apollo'
import { Link } from 'redux-little-router'
import Github from 'src/app/github/github'
import { setUser } from 'src/reducers/users'

class App extends React.Component {
  componentDidUpdate() {
    const { data, setUser, user } = this.props
    if (data
      && data.currentPerson
      && (!user || data.currentPerson.id !== user.id)) {
      setUser(data.currentPerson)
     }
  }

  render () {
    const { data, user } = this.props
    if (data.loading) {
      return <div className='loading-text'></div>
    }
    if (!user) {
      return <Link className='button' href='/sign-in'>Sign in</Link>
    }

    const GithubComponent = Github(user.githubAuth)
    return (
      <div className='section main'>
        <div className='container'>
          <GithubComponent />
        </div>
      </div>
    )
  }
}

export const QUERY = gql`
{
  currentPerson {
    fullName
    githubAuth
    id
  }
}
`

App.propTypes = {
  loading: PropTypes.bool,
  user: PropTypes.shape({
    fullName: PropTypes.string,
    githubAuth: PropTypes.string
  })
}

export default connect(
  ({ tokens, users }) => ({ token: tokens.token, user: users.user }),
  ({
    setUser
  })
)(graphql(QUERY)(App))
