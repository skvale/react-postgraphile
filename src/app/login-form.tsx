import React from 'react'
import { useMachine } from '@xstate/react'
import { print } from 'graphql'
import Cookies from 'js-cookie'
import { useQueryParam, BooleanParam } from 'use-query-params'
import { loader } from 'graphql.macro'
import { Button } from '../components/button'
import { Input } from '../components/input'
import { Form } from '../components/form'
import { Person } from '../schema'
import { Link } from '../components/link'
import { Card } from '../components/card'
import {
  loginMachine,
  registerMachine,
  LoginContext
} from '../machines/login-machine'
import { createClient } from '../clients/postgraphile'

const registerPersonMutation = loader('./graphql/register-person-mutation.gql')
const authenticateMutation = loader('./graphql/authenticate-mutation.gql')

const login = (...args: any[]) =>
  createClient()(print(authenticateMutation))(...args)
const register = (...args: any[]) =>
  createClient()(print(registerPersonMutation))(...args)

export type LoginFormProps = {
  currentUser?: Person | null
  updateToken: (jwtToken: string) => void
}

enum ViewState {
  LOGIN = 'login',
  REGISTER = 'register'
}

export const LoginForm: React.FC<LoginFormProps> = ({
  currentUser,
  updateToken
}) => {
  const [registerParam] = useQueryParam('register', BooleanParam)

  const [current, send] = useMachine(
    registerParam ? registerMachine : loginMachine,
    {
      services: {
        onLogin: async ({ email, password }: LoginContext) => {
          const result = await login({
            email,
            password
          })
          console.log(result.authenticate)
          if (result && result.authenticate && result.authenticate.jwtToken) {
            Cookies.set('postgraphile-jwt', result.authenticate.jwtToken)
            updateToken(result.authenticate.jwtToken)
            return result.authenticate.jwtToken
          } else {
            throw Error('something happened')
          }
        },
        onRegister: async ({
          email,
          password,
          firstName,
          lastName
        }: LoginContext) => {
          return register({
            email,
            firstName,
            lastName,
            password
          })
        }
      }
    }
  ) as any

  const { error, firstName, email, lastName, password } = current.context

  if (current.matches('success')) {
    return null
  }

  const onSwapState = () => {
    send('TOGGLE')
  }

  return (
    <div className='flex justify-center'>
      {currentUser && <div>Currently logged in as: {currentUser.fullName}</div>}
      <Card
        className='bg-white'
        title={current.matches(ViewState.LOGIN) ? 'Login' : 'Register'}
      >
        <div className='float-right'>
          {current.matches(ViewState.LOGIN) ? (
            <Link data-testid='toggle-link' color='gray' onClick={onSwapState}>
              Register
            </Link>
          ) : (
            <Link data-testid='toggle-link' color='gray' onClick={onSwapState}>
              Login
            </Link>
          )}
        </div>
        <Form
          data-testid='login-form'
          error={error}
          onSubmit={(e: React.FormEvent<HTMLFormElement>) => {
            send({
              type: 'SUBMIT'
            })
          }}
        >
          <div className='mb-6'>
            <Input
              data-testid='login-email'
              id='email'
              label='Email'
              onChange={e => send({ type: 'EMAIL', value: e.target.value })}
              value={email}
            />
            <Input
              data-testid='login-password'
              id='password'
              label='Password'
              type='password'
              onChange={e => send({ type: 'PASSWORD', value: e.target.value })}
              value={password}
            />
            {current.matches(ViewState.REGISTER) && (
              <React.Fragment>
                <Input
                  data-testid='login-first-name'
                  id='firstName'
                  label='First Name'
                  onChange={e =>
                    send({ type: 'FIRST_NAME', value: e.target.value })
                  }
                  value={firstName}
                />
                <Input
                  data-testid='login-last-name'
                  id='lastName'
                  label='Last Name'
                  onChange={e =>
                    send({ type: 'LAST_NAME', value: e.target.value })
                  }
                  value={lastName}
                />
              </React.Fragment>
            )}
          </div>
          <div className='flex items-center justify-between'>
            <Button data-testid='login-submit-button' type='submit'>
              {current.matches(ViewState.LOGIN) ? 'Sign In' : 'Register'}
            </Button>
          </div>
        </Form>
      </Card>
    </div>
  )
}
