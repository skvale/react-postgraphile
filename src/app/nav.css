.navbar {
  overflow: hidden;
}

.nav-user-icon {
  margin-top: -4px;
  margin-right: 4px;
}

.nav-user {
  margin-top: 6px;
  margin-right: 8px;
  z-index: 2;
}

.nav-user-signout {
  margin-top: 6px;
  margin-right: 8px;
}

a.navbar-item {
  padding-left: 0;
}

.nav-logo {
  background: #444;
  color: #eee;
  border-radius: 50%;
  left: 24px;
  padding: 4px;
  margin-left: 16px;
  transition: 2s;
  z-index: 2;
}

.nav-logo:hover {
  transform: rotate(720deg);
}

.nav-title {
  background: white;
  left: 72px;
  padding: 4px 16px;
  border-radius: 3px;
  z-index: 2;
  margin-left: 16px;
}

.nav-logo-texture {
  background-color: white;
}

.navbar-rough-canvas {
  position: absolute;
}