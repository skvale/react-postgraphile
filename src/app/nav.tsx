import React, { useState, useEffect, useRef } from 'react'
import rough from 'roughjs/dist/rough.umd'
import { Person } from '../schema'

export type NavProps = {
  currentUser?: Person | null
}

export const Nav: React.FC<NavProps> = ({ currentUser }) => {
  const [showPullout, setShowPullout] = useState(false)
  const svgEl = useRef<SVGSVGElement>(null)
  useEffect(() => {
    if (svgEl && svgEl.current) {
      const rc = rough.svg(svgEl.current)
      let node = rc.rectangle(10, 10, 40, 40, {
        fill: 'blue'
      })
      svgEl.current.appendChild(node)
    }
  })

  return (
    <div>
      <div className='container mx-auto px-4'>
        <div className='flex items-center justify-between'>
          <div>
            <a href='/'>
              <svg ref={svgEl} className='h-10 w-10' viewBox='0 0 55 55' />
            </a>
          </div>
          <div className='hidden sm:flex sm:items-center'>
            <a
              href='/products'
              className='text-gray-800 text-sm font-semibold hover:text-purple-600 mr-4'
            >
              Products
            </a>
            {currentUser ? (
              <a
                href='/sign-out'
                className='bg-gray-900 text-white text-sm font-semibold border px-4 py-2 rounded-lg hover:text-gray-200 hover:border-blue-600'
              >
                Sign out
              </a>
            ) : (
              <React.Fragment>
                <a
                  data-testid='signin-link'
                  href='/login'
                  className='text-gray-800 text-sm font-semibold hover:text-purple-600 mr-4'
                >
                  Sign in
                </a>
                <a
                  href='/login?register=1'
                  className='bg-gray-900 text-white text-sm font-semibold border px-4 py-2 rounded-lg hover:text-gray-200 hover:border-blue-600'
                >
                  Sign up
                </a>
              </React.Fragment>
            )}
          </div>

          <div
            className='sm:hidden cursor-pointer'
            onClick={() => setShowPullout(!showPullout)}
          >
            <svg className='w-6 h-6 text-purple-600' viewBox='0 0 24 24'>
              <path
                fill='currentColor'
                d='M12.9499909,17 C12.7183558,18.1411202 11.709479,19 10.5,19 C9.29052104,19 8.28164422,18.1411202 8.05000906,17 L3.5,17 C3.22385763,17 3,16.7761424 3,16.5 C3,16.2238576 3.22385763,16 3.5,16 L8.05000906,16 C8.28164422,14.8588798 9.29052104,14 10.5,14 C11.709479,14 12.7183558,14.8588798 12.9499909,16 L20.5,16 C20.7761424,16 21,16.2238576 21,16.5 C21,16.7761424 20.7761424,17 20.5,17 L12.9499909,17 Z M18.9499909,12 C18.7183558,13.1411202 17.709479,14 16.5,14 C15.290521,14 14.2816442,13.1411202 14.0500091,12 L3.5,12 C3.22385763,12 3,11.7761424 3,11.5 C3,11.2238576 3.22385763,11 3.5,11 L14.0500091,11 C14.2816442,9.85887984 15.290521,9 16.5,9 C17.709479,9 18.7183558,9.85887984 18.9499909,11 L20.5,11 C20.7761424,11 21,11.2238576 21,11.5 C21,11.7761424 20.7761424,12 20.5,12 L18.9499909,12 Z M9.94999094,7 C9.71835578,8.14112016 8.70947896,9 7.5,9 C6.29052104,9 5.28164422,8.14112016 5.05000906,7 L3.5,7 C3.22385763,7 3,6.77614237 3,6.5 C3,6.22385763 3.22385763,6 3.5,6 L5.05000906,6 C5.28164422,4.85887984 6.29052104,4 7.5,4 C8.70947896,4 9.71835578,4.85887984 9.94999094,6 L20.5,6 C20.7761424,6 21,6.22385763 21,6.5 C21,6.77614237 20.7761424,7 20.5,7 L9.94999094,7 Z M7.5,8 C8.32842712,8 9,7.32842712 9,6.5 C9,5.67157288 8.32842712,5 7.5,5 C6.67157288,5 6,5.67157288 6,6.5 C6,7.32842712 6.67157288,8 7.5,8 Z M16.5,13 C17.3284271,13 18,12.3284271 18,11.5 C18,10.6715729 17.3284271,10 16.5,10 C15.6715729,10 15,10.6715729 15,11.5 C15,12.3284271 15.6715729,13 16.5,13 Z M10.5,18 C11.3284271,18 12,17.3284271 12,16.5 C12,15.6715729 11.3284271,15 10.5,15 C9.67157288,15 9,15.6715729 9,16.5 C9,17.3284271 9.67157288,18 10.5,18 Z'
              />
            </svg>
          </div>
        </div>

        <div
          className={`block sm:hidden bg-white border-t-2 py-2${
            showPullout ? ' hidden' : ''
          }`}
        >
          <div className='flex flex-col'>
            <a
              href='/products'
              className='text-gray-800 text-sm font-semibold hover:text-purple-600 mb-1'
            >
              Products
            </a>
            <div className='flex justify-between items-center border-t-2 pt-2'>
              {currentUser ? (
                <a
                  href='/sign-out'
                  className='text-gray-800 text-sm font-semibold border px-4 py-1 rounded-lg hover:text-purple-600 hover:border-purple-600'
                >
                  Sign out
                </a>
              ) : (
                <React.Fragment>
                  <a
                    data-testid='signin-link'
                    href='/login'
                    className='text-gray-800 text-sm font-semibold hover:text-purple-600 mr-4'
                  >
                    Sign in
                  </a>
                  <a
                    href='/login?register=true'
                    className='text-gray-800 text-sm font-semibold border px-4 py-1 rounded-lg hover:text-purple-600 hover:border-purple-600'
                  >
                    Sign up
                  </a>
                </React.Fragment>
              )}
            </div>
          </div>
        </div>
      </div>
    </div>
  )
}
