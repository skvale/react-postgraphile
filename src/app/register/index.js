import React from 'react'
import PropTypes from 'prop-types'
import { graphql } from 'react-apollo'
import { connect } from 'react-redux'
import gql from 'graphql-tag'
import { push } from 'redux-little-router'
import { setCookie } from 'src/utils/cookies'
import { setToken } from 'src/reducers/tokens'
import { noAuthClient } from 'src/clients/postgraphile'

import SignUp from './sign-up'
import SignIn from './sign-in'

class Registration extends React.Component {
  state = {
    active: 'SIGN_IN'
  }

  signIn = input => this.props.authenticate(input).then(this.handleNewToken)

  handleNewToken = ({ data }) => {
    const { jwtToken } = data.authenticate
    if (jwtToken) {
      setCookie('jwt_token', jwtToken)
      this.props.setToken(jwtToken)
      this.props.navigateToHome()
    } else {
      this.setState({
        invalidCredentials: true
      })
    }
  }

  toggle = () => {
    this.setState(prevState => ({
      active: prevState.active === 'SIGN_IN' ? 'SIGN_UP' : 'SIGN_IN',
      invalidCredentials: false
    }))
  }

  render () {
    return (
      <div>
        <div className='section'>
          {this.state.active === 'SIGN_IN' ? (
            <SignIn signIn={this.signIn} />
          ) : (
            <SignUp signIn={this.signIn} />
          )}
          <p className='help is-danger'>
            {this.state.invalidCredentials && ' * invalid login'}
          </p>
          <div>or</div>
          <button className='button' onClick={this.toggle}>
            {this.state.active === 'SIGN_IN'
              ? "I'm a new user"
              : 'I have an account'}
          </button>
        </div>
      </div>
    )
  }
}

Registration.propTypes = {
  authenticate: PropTypes.func.isRequired,
  navigateToHome: PropTypes.func.isRequired,
  setToken: PropTypes.func.isRequired
}

const ApolloRegistration = graphql(
  gql`
    mutation Authenticate($email: String!, $password: String!) {
      authenticate(input: { email: $email, password: $password }) {
        jwtToken
      }
    }
  `,
  {
    options: {
      client: noAuthClient
    },
    props: ({ mutate }) => ({
      authenticate: variables => mutate({ variables })
    })
  }
)(Registration)

export default connect(
  null,
  dispatch => ({
    navigateToHome: () => dispatch(push('/')),
    setToken: token => dispatch(setToken(token))
  })
)(ApolloRegistration)
