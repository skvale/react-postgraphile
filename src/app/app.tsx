import React, { useState } from 'react'
import { loader } from 'graphql.macro'
import { print } from 'graphql'
import Cookies from 'js-cookie'
import { Routing } from './routing'
import { Nav } from './nav'
import { createClient } from '../clients/postgraphile'
import { useAsyncEffect } from '../utils'

const currentPersonQuery = loader('./graphql/current-person-query.gql')

export type AppProps = {}

export const App: React.FC<AppProps> = () => {
  const [token, setToken] = useState<string>(
    Cookies.get('postgraphile-jwt') || ''
  )
  const [currentUser, setUser] = useState()
  const [error, setError] = useState()
  const [loading, setLoading] = useState(false)

  useAsyncEffect(
    async signal => {
      const query = print(currentPersonQuery)
      const getCurrentUser = createClient(token)(query)
      setLoading(true)
      try {
        const currentUser = await getCurrentUser()
        if (!signal.aborted) {
          setUser(currentUser.currentPerson)
        }
      } catch (e) {
        setError(e)
      } finally {
        setLoading(false)
      }
    },
    [token]
  )

  if (error) {
    return <div>Error: {error.message}</div>
  }
  return (
    <div className='bg-gray-100 font-sans w-full min-h-screen m-0'>
      <Nav currentUser={currentUser} />
      <Routing
        currentUser={currentUser}
        loading={loading}
        updateToken={setToken}
      />
    </div>
  )
}
