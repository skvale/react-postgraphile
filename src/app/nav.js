import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { Link } from "redux-little-router";
import Connected from "react-icons/lib/fa/connectdevelop";
import SignOut from "react-icons/lib/fa/sign-out";
import Person from "react-icons/lib/fa/user";
import rough from "roughjs";

import "./nav.css";

const colors = {
  BRAND_MAIN: "#55ee429",
  BRAND_SUB: "#9933ea"
};

class Nav extends React.Component {
  componentDidMount() {
    const rc = rough.canvas(this.canvas);
    rc.rectangle(5, 5, 210, 45, { fill: colors.BRAND_MAIN });
    rc.rectangle(225, 5, 70, 45, { fill: colors.BRAND_SUB });
  }

  render() {
    const { user } = this.props;
    return (
      <nav className='navbar'>
        <div className='navbar-brand'>
          <Link className='navbar-item' href='/' title='Home'>
            <Connected className='icon is-medium logo nav-logo' />
            <h1 className='nav-title'>Pull requests</h1>
          </Link>
          <canvas
            className='rough-canvas'
            ref={canvas => {
              this.canvas = canvas;
            }}
          />
        </div>
        <div className='navbar-menu'>
          {user && (
            <div className='navbar-end'>
              <Link className='button nav-user' href='/profile' title='Profile'>
                <Person className='icon nav-user-icon' />
                <span className='nav-user-name'>{user.fullName}</span>
              </Link>
              <Link
                className='button nav-user-signout'
                title='Sign out'
                href='/sign-out'
              >
                <span>Sign out</span>
                <SignOut className='icon' />
              </Link>
            </div>
          )}
        </div>
      </nav>
    );
  }
}

Nav.propTypes = {
  user: PropTypes.shape({
    fullName: PropTypes.string
  })
};

export default connect(({ users }) => ({ user: users.user }))(Nav);
