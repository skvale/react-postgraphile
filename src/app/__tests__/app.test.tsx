import '@testing-library/jest-dom/extend-expect'

import React from 'react'
import { render } from '@testing-library/react'
import { App } from '../app'
import { LoginFormProps } from '../login-form'
import rough from 'roughjs/dist/rough.umd'
import * as graph from '../../clients/postgraphile'

rough.svg = jest.fn(() => ({ rectangle: () => document.createElement('svg') }))

function flushPromises() {
  return new Promise(resolve => setImmediate(resolve))
}

jest.mock('../login-form', () => ({
  LoginForm: (props: LoginFormProps) => `LoginForm ${JSON.stringify(props)}`
}))

test('renders without user', async () => {
  const m = jest.fn(() => Promise.resolve({ currentPerson: null }))
  const f = jest.fn(() => m)
  // @ts-ignore
  graph.createClient = () => f
  const { asFragment } = render(<App />)
  await flushPromises()

  expect(m).toHaveBeenCalled()
  expect(f.mock.calls[0]).toMatchSnapshot()
  expect(asFragment()).toMatchSnapshot()
})

test('renders with user', async () => {
  const m = jest.fn(() =>
    Promise.resolve({ currentPerson: { fullName: 'Bobby' } })
  )
  const f = jest.fn(() => m)
  // @ts-ignore
  graph.createClient = () => f
  const { asFragment } = render(<App />)
  await flushPromises()
  expect(asFragment()).toMatchSnapshot()
})

test('renders loading', () => {
  const m = jest.fn(() =>
    Promise.resolve({ currentPerson: { fullName: 'Bobby' } })
  )
  const f = jest.fn(() => m)
  // @ts-ignore
  graph.createClient = () => f
  const { asFragment } = render(<App />)
  expect(asFragment()).toMatchSnapshot()
})

test('renders error', async () => {
  const m = jest.fn(() => Promise.reject(Error('bad stuff')))
  const f = jest.fn(() => m)
  // @ts-ignore
  graph.createClient = () => f
  const { asFragment, rerender } = render(<App />)
  await flushPromises()
  rerender(<App />)

  expect(asFragment()).toMatchSnapshot()
})
