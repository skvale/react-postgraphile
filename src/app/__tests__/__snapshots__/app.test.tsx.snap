// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders error 1`] = `
<DocumentFragment>
  <div>
    Error: bad stuff
  </div>
</DocumentFragment>
`;

exports[`renders loading 1`] = `
<DocumentFragment>
  <div
    class="bg-gray-100 font-sans w-full min-h-screen m-0"
  >
    <div>
      <div
        class="container mx-auto px-4"
      >
        <div
          class="flex items-center justify-between"
        >
          <div>
            <a
              href="/"
            >
              <svg
                class="h-10 w-10"
                viewBox="0 0 55 55"
              >
                <svg />
                <svg />
              </svg>
            </a>
          </div>
          <div
            class="hidden sm:flex sm:items-center"
          >
            <a
              class="text-gray-800 text-sm font-semibold hover:text-purple-600 mr-4"
              href="/products"
            >
              Products
            </a>
            <a
              class="text-gray-800 text-sm font-semibold hover:text-purple-600 mr-4"
              data-testid="signin-link"
              href="/login"
            >
              Sign in
            </a>
            <a
              class="bg-gray-900 text-white text-sm font-semibold border px-4 py-2 rounded-lg hover:text-gray-200 hover:border-blue-600"
              href="/login?register=1"
            >
              Sign up
            </a>
          </div>
          <div
            class="sm:hidden cursor-pointer"
          >
            <svg
              class="w-6 h-6 text-purple-600"
              viewBox="0 0 24 24"
            >
              <path
                d="M12.9499909,17 C12.7183558,18.1411202 11.709479,19 10.5,19 C9.29052104,19 8.28164422,18.1411202 8.05000906,17 L3.5,17 C3.22385763,17 3,16.7761424 3,16.5 C3,16.2238576 3.22385763,16 3.5,16 L8.05000906,16 C8.28164422,14.8588798 9.29052104,14 10.5,14 C11.709479,14 12.7183558,14.8588798 12.9499909,16 L20.5,16 C20.7761424,16 21,16.2238576 21,16.5 C21,16.7761424 20.7761424,17 20.5,17 L12.9499909,17 Z M18.9499909,12 C18.7183558,13.1411202 17.709479,14 16.5,14 C15.290521,14 14.2816442,13.1411202 14.0500091,12 L3.5,12 C3.22385763,12 3,11.7761424 3,11.5 C3,11.2238576 3.22385763,11 3.5,11 L14.0500091,11 C14.2816442,9.85887984 15.290521,9 16.5,9 C17.709479,9 18.7183558,9.85887984 18.9499909,11 L20.5,11 C20.7761424,11 21,11.2238576 21,11.5 C21,11.7761424 20.7761424,12 20.5,12 L18.9499909,12 Z M9.94999094,7 C9.71835578,8.14112016 8.70947896,9 7.5,9 C6.29052104,9 5.28164422,8.14112016 5.05000906,7 L3.5,7 C3.22385763,7 3,6.77614237 3,6.5 C3,6.22385763 3.22385763,6 3.5,6 L5.05000906,6 C5.28164422,4.85887984 6.29052104,4 7.5,4 C8.70947896,4 9.71835578,4.85887984 9.94999094,6 L20.5,6 C20.7761424,6 21,6.22385763 21,6.5 C21,6.77614237 20.7761424,7 20.5,7 L9.94999094,7 Z M7.5,8 C8.32842712,8 9,7.32842712 9,6.5 C9,5.67157288 8.32842712,5 7.5,5 C6.67157288,5 6,5.67157288 6,6.5 C6,7.32842712 6.67157288,8 7.5,8 Z M16.5,13 C17.3284271,13 18,12.3284271 18,11.5 C18,10.6715729 17.3284271,10 16.5,10 C15.6715729,10 15,10.6715729 15,11.5 C15,12.3284271 15.6715729,13 16.5,13 Z M10.5,18 C11.3284271,18 12,17.3284271 12,16.5 C12,15.6715729 11.3284271,15 10.5,15 C9.67157288,15 9,15.6715729 9,16.5 C9,17.3284271 9.67157288,18 10.5,18 Z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
        <div
          class="block sm:hidden bg-white border-t-2 py-2"
        >
          <div
            class="flex flex-col"
          >
            <a
              class="text-gray-800 text-sm font-semibold hover:text-purple-600 mb-1"
              href="/products"
            >
              Products
            </a>
            <div
              class="flex justify-between items-center border-t-2 pt-2"
            >
              <a
                class="text-gray-800 text-sm font-semibold hover:text-purple-600 mr-4"
                data-testid="signin-link"
                href="/login"
              >
                Sign in
              </a>
              <a
                class="text-gray-800 text-sm font-semibold border px-4 py-1 rounded-lg hover:text-purple-600 hover:border-purple-600"
                href="/login?register=true"
              >
                Sign up
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      loading
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders with user 1`] = `
<DocumentFragment>
  <div
    class="bg-gray-100 font-sans w-full min-h-screen m-0"
  >
    <div>
      <div
        class="container mx-auto px-4"
      >
        <div
          class="flex items-center justify-between"
        >
          <div>
            <a
              href="/"
            >
              <svg
                class="h-10 w-10"
                viewBox="0 0 55 55"
              >
                <svg />
                <svg />
                <svg />
                <svg />
              </svg>
            </a>
          </div>
          <div
            class="hidden sm:flex sm:items-center"
          >
            <a
              class="text-gray-800 text-sm font-semibold hover:text-purple-600 mr-4"
              href="/products"
            >
              Products
            </a>
            <a
              class="bg-gray-900 text-white text-sm font-semibold border px-4 py-2 rounded-lg hover:text-gray-200 hover:border-blue-600"
              href="/sign-out"
            >
              Sign out
            </a>
          </div>
          <div
            class="sm:hidden cursor-pointer"
          >
            <svg
              class="w-6 h-6 text-purple-600"
              viewBox="0 0 24 24"
            >
              <path
                d="M12.9499909,17 C12.7183558,18.1411202 11.709479,19 10.5,19 C9.29052104,19 8.28164422,18.1411202 8.05000906,17 L3.5,17 C3.22385763,17 3,16.7761424 3,16.5 C3,16.2238576 3.22385763,16 3.5,16 L8.05000906,16 C8.28164422,14.8588798 9.29052104,14 10.5,14 C11.709479,14 12.7183558,14.8588798 12.9499909,16 L20.5,16 C20.7761424,16 21,16.2238576 21,16.5 C21,16.7761424 20.7761424,17 20.5,17 L12.9499909,17 Z M18.9499909,12 C18.7183558,13.1411202 17.709479,14 16.5,14 C15.290521,14 14.2816442,13.1411202 14.0500091,12 L3.5,12 C3.22385763,12 3,11.7761424 3,11.5 C3,11.2238576 3.22385763,11 3.5,11 L14.0500091,11 C14.2816442,9.85887984 15.290521,9 16.5,9 C17.709479,9 18.7183558,9.85887984 18.9499909,11 L20.5,11 C20.7761424,11 21,11.2238576 21,11.5 C21,11.7761424 20.7761424,12 20.5,12 L18.9499909,12 Z M9.94999094,7 C9.71835578,8.14112016 8.70947896,9 7.5,9 C6.29052104,9 5.28164422,8.14112016 5.05000906,7 L3.5,7 C3.22385763,7 3,6.77614237 3,6.5 C3,6.22385763 3.22385763,6 3.5,6 L5.05000906,6 C5.28164422,4.85887984 6.29052104,4 7.5,4 C8.70947896,4 9.71835578,4.85887984 9.94999094,6 L20.5,6 C20.7761424,6 21,6.22385763 21,6.5 C21,6.77614237 20.7761424,7 20.5,7 L9.94999094,7 Z M7.5,8 C8.32842712,8 9,7.32842712 9,6.5 C9,5.67157288 8.32842712,5 7.5,5 C6.67157288,5 6,5.67157288 6,6.5 C6,7.32842712 6.67157288,8 7.5,8 Z M16.5,13 C17.3284271,13 18,12.3284271 18,11.5 C18,10.6715729 17.3284271,10 16.5,10 C15.6715729,10 15,10.6715729 15,11.5 C15,12.3284271 15.6715729,13 16.5,13 Z M10.5,18 C11.3284271,18 12,17.3284271 12,16.5 C12,15.6715729 11.3284271,15 10.5,15 C9.67157288,15 9,15.6715729 9,16.5 C9,17.3284271 9.67157288,18 10.5,18 Z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
        <div
          class="block sm:hidden bg-white border-t-2 py-2"
        >
          <div
            class="flex flex-col"
          >
            <a
              class="text-gray-800 text-sm font-semibold hover:text-purple-600 mb-1"
              href="/products"
            >
              Products
            </a>
            <div
              class="flex justify-between items-center border-t-2 pt-2"
            >
              <a
                class="text-gray-800 text-sm font-semibold border px-4 py-1 rounded-lg hover:text-purple-600 hover:border-purple-600"
                href="/sign-out"
              >
                Sign out
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div>
        Hello: Bobby
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders without user 1`] = `
Array [
  "query currentPerson {
  currentPerson {
    fullName
    githubAuth
    id
  }
}
",
]
`;

exports[`renders without user 2`] = `
<DocumentFragment>
  <div
    class="bg-gray-100 font-sans w-full min-h-screen m-0"
  >
    <div>
      <div
        class="container mx-auto px-4"
      >
        <div
          class="flex items-center justify-between"
        >
          <div>
            <a
              href="/"
            >
              <svg
                class="h-10 w-10"
                viewBox="0 0 55 55"
              >
                <svg />
                <svg />
                <svg />
              </svg>
            </a>
          </div>
          <div
            class="hidden sm:flex sm:items-center"
          >
            <a
              class="text-gray-800 text-sm font-semibold hover:text-purple-600 mr-4"
              href="/products"
            >
              Products
            </a>
            <a
              class="text-gray-800 text-sm font-semibold hover:text-purple-600 mr-4"
              data-testid="signin-link"
              href="/login"
            >
              Sign in
            </a>
            <a
              class="bg-gray-900 text-white text-sm font-semibold border px-4 py-2 rounded-lg hover:text-gray-200 hover:border-blue-600"
              href="/login?register=1"
            >
              Sign up
            </a>
          </div>
          <div
            class="sm:hidden cursor-pointer"
          >
            <svg
              class="w-6 h-6 text-purple-600"
              viewBox="0 0 24 24"
            >
              <path
                d="M12.9499909,17 C12.7183558,18.1411202 11.709479,19 10.5,19 C9.29052104,19 8.28164422,18.1411202 8.05000906,17 L3.5,17 C3.22385763,17 3,16.7761424 3,16.5 C3,16.2238576 3.22385763,16 3.5,16 L8.05000906,16 C8.28164422,14.8588798 9.29052104,14 10.5,14 C11.709479,14 12.7183558,14.8588798 12.9499909,16 L20.5,16 C20.7761424,16 21,16.2238576 21,16.5 C21,16.7761424 20.7761424,17 20.5,17 L12.9499909,17 Z M18.9499909,12 C18.7183558,13.1411202 17.709479,14 16.5,14 C15.290521,14 14.2816442,13.1411202 14.0500091,12 L3.5,12 C3.22385763,12 3,11.7761424 3,11.5 C3,11.2238576 3.22385763,11 3.5,11 L14.0500091,11 C14.2816442,9.85887984 15.290521,9 16.5,9 C17.709479,9 18.7183558,9.85887984 18.9499909,11 L20.5,11 C20.7761424,11 21,11.2238576 21,11.5 C21,11.7761424 20.7761424,12 20.5,12 L18.9499909,12 Z M9.94999094,7 C9.71835578,8.14112016 8.70947896,9 7.5,9 C6.29052104,9 5.28164422,8.14112016 5.05000906,7 L3.5,7 C3.22385763,7 3,6.77614237 3,6.5 C3,6.22385763 3.22385763,6 3.5,6 L5.05000906,6 C5.28164422,4.85887984 6.29052104,4 7.5,4 C8.70947896,4 9.71835578,4.85887984 9.94999094,6 L20.5,6 C20.7761424,6 21,6.22385763 21,6.5 C21,6.77614237 20.7761424,7 20.5,7 L9.94999094,7 Z M7.5,8 C8.32842712,8 9,7.32842712 9,6.5 C9,5.67157288 8.32842712,5 7.5,5 C6.67157288,5 6,5.67157288 6,6.5 C6,7.32842712 6.67157288,8 7.5,8 Z M16.5,13 C17.3284271,13 18,12.3284271 18,11.5 C18,10.6715729 17.3284271,10 16.5,10 C15.6715729,10 15,10.6715729 15,11.5 C15,12.3284271 15.6715729,13 16.5,13 Z M10.5,18 C11.3284271,18 12,17.3284271 12,16.5 C12,15.6715729 11.3284271,15 10.5,15 C9.67157288,15 9,15.6715729 9,16.5 C9,17.3284271 9.67157288,18 10.5,18 Z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
        <div
          class="block sm:hidden bg-white border-t-2 py-2"
        >
          <div
            class="flex flex-col"
          >
            <a
              class="text-gray-800 text-sm font-semibold hover:text-purple-600 mb-1"
              href="/products"
            >
              Products
            </a>
            <div
              class="flex justify-between items-center border-t-2 pt-2"
            >
              <a
                class="text-gray-800 text-sm font-semibold hover:text-purple-600 mr-4"
                data-testid="signin-link"
                href="/login"
              >
                Sign in
              </a>
              <a
                class="text-gray-800 text-sm font-semibold border px-4 py-1 rounded-lg hover:text-purple-600 hover:border-purple-600"
                href="/login?register=true"
              >
                Sign up
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div>
        Not logged in
      </div>
    </div>
  </div>
</DocumentFragment>
`;
